dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/trackedentity/AttributeValidator.java

```diff
@@ -94,29 +95,31 @@ private void validateMandatoryAttributes(
       TrackerBundle bundle,
       org.hisp.dhis.tracker.imports.domain.TrackedEntity trackedEntity,
       TrackedEntityType trackedEntityType) {
-    if (trackedEntityType != null) {
-      Set<MetadataIdentifier> trackedEntityAttributes =
-          trackedEntity.getAttributes().stream()
-              .map(Attribute::getAttribute)
-              .collect(Collectors.toSet());
-
-      TrackerIdSchemeParams idSchemes = bundle.getPreheat().getIdSchemes();
-      trackedEntityType.getTrackedEntityTypeAttributes().stream()
-          .filter(
-              trackedEntityTypeAttribute ->
-                  Boolean.TRUE.equals(trackedEntityTypeAttribute.isMandatory()))
-          .map(TrackedEntityTypeAttribute::getTrackedEntityAttribute)
-          .map(idSchemes::toMetadataIdentifier)
-          .filter(mandatoryAttribute -> !trackedEntityAttributes.contains(mandatoryAttribute))
-          .forEach(
-              attribute ->
-                  reporter.addError(
-                      trackedEntity,
-                      E1090,
-                      attribute,
-                      trackedEntityType.getUid(),
-                      trackedEntity.getTrackedEntity()));
+    if (bundle.getStrategy(trackedEntity) == UPDATE) {
+      return;
```

Comment:
> Always validate that no mandatory attribute is being deleted

Where does that take place?

Reply:
Same here, the null is checked in `validateAttributes` method